qiskit/utils/quantum_instance.py

```diff
@@ -364,19 +364,23 @@ def assemble(self, circuits) -> Qobj:
 
         return compiler.assemble(circuits, **self._run_config.to_dict())
 
-    def execute(self, circuits, had_transpiled: bool = False):
+    # to-do : add max - tries param
+    def execute(self, circuits, had_transpiled: bool = False, max_tries=50):
```

Comment:
I was envisioning a parameter on the constructor so that a user, when creating the QI, could change the default. While logically it is changeable via this parameter here, algorithms and opflow directly call through to execute here to run/execute the circuits they build in an environment (backend plus config, for transpiler etc etc) that can be defined by the user wanting to run the algorithm. Otherwise they have no way to influence this max tries value as the calls to execute etc are deep inside the algorithms/opflow.

Reply:
My motivation was that if a user has defined a particular `QuantumInstance`, they would be able to execute different job sets with a different number of trials (if they needed to do that) without having to make a new object or changing the attribute value of the object.
I think what we may do is define `max_tries` parameters in the `execute` method too. If the user explicitly specifies the trials in the `execute` function, it overrides the original attribute of the object, else it runs with the original trials attribute of the object. What do you think ?

Reply:
With the algorithms the user does not execute  anything the algorithm does. The user runs the algorithm, eg calls its solve, find_min_eigvalue or whatever the method is named that given a problem the algorithm runs/executes quantum circuits to return a result. Such configuration over the execution environment was put in one place in the QuantumInstance and that is what an algorithm receives. Otherwise all of that info logically has to be configurable on the algorithms to allow contrl over how its circuits are transpilied, executed etc etc. Yes if some uses the code more directly and calls execute themsleves they would, how you have done it, have finer grain control. Maybe in the future we will want algorithms to be more adaptive in that regard to say do things differently as they near convergence, and in which case that config may end up being reflected in the algorithm in some way. Today that does not happen - you can see the set of params that QI has on its constructor - given an algorithm thats how you as end user of an algorithm have some control over its execution etc.